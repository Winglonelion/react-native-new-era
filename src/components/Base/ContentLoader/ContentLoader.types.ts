import { SvgProps } from 'react-native-svg';

export interface IContentLoaderProps extends SvgProps {
  animate?: boolean;
  backgroundColor?: string;
  foregroundColor?: string;
  rtl?: boolean;
  speed?: number;
  interval?: number;
  uniqueKey?: string;
}
