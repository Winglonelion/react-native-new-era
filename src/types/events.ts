import { GestureResponderEvent } from 'react-native';

export type PressEvent = (event: GestureResponderEvent) => any;
