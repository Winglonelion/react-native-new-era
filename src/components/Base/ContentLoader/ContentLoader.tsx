import * as React from 'react';
import { Circle, Path, Rect } from 'react-native-svg';

import { IContentLoaderProps } from './ContentLoader.types';
import ReactContentLoaderFacebook from './presets/FacebookStyle';
import NativeSvg from './Svg';

const ContentLoader: React.FC<IContentLoaderProps> = props =>
  props.children ? (
    <NativeSvg {...props} />
  ) : (
    <ReactContentLoaderFacebook {...props} />
  );

export { Circle, Rect, Path };

export default ContentLoader;
