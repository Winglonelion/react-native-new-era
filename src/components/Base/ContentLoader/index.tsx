import ContentLoader from './ContentLoader';

export { Circle, Rect, Path } from './ContentLoader';
export default ContentLoader;
