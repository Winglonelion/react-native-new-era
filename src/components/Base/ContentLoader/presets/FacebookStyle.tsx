import * as React from 'react';

import { Circle, Rect } from 'react-native-svg';
import { IContentLoaderProps } from '../ContentLoader.types';
import NativeSvg from '../Svg';

const ReactContentLoaderFacebook: React.FC<IContentLoaderProps> = props => (
  <NativeSvg viewBox="0 0 476 124" width={476} height={124} {...props}>
    <Rect x="48" y="8" width="88" height="6" rx="3" />
    <Rect x="48" y="26" width="52" height="6" rx="3" />
    <Rect x="0" y="56" width="410" height="6" rx="3" />
    <Rect x="0" y="72" width="380" height="6" rx="3" />
    <Rect x="0" y="88" width="178" height="6" rx="3" />
    <Circle cx="20" cy="20" r="20" />
  </NativeSvg>
);

export default ReactContentLoaderFacebook;
