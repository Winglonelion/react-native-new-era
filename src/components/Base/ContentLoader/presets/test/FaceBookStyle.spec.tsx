import React from 'react';
import { shallow } from 'enzyme';
import FacebookStyle from '../FacebookStyle';
import { IContentLoaderProps } from '../../ContentLoader.types';

const setup = (
  propOverrides: IContentLoaderProps = {} as IContentLoaderProps,
) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<FacebookStyle {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('FacebookStyle check snapshot', () => {
  test('should match snapshot', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
