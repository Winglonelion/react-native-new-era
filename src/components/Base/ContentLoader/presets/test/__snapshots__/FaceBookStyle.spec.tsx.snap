// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookStyle check snapshot should match snapshot 1`] = `
<NativeSvg
  animate={true}
  backgroundColor="#f5f6f7"
  foregroundColor="#eee"
  height={124}
  interval={0.25}
  rtl={false}
  speed={1.2}
  style={Object {}}
  viewBox="0 0 476 124"
  width={476}
>
  <Rect
    height="6"
    rx="3"
    width="88"
    x="48"
    y="8"
  />
  <Rect
    height="6"
    rx="3"
    width="52"
    x="48"
    y="26"
  />
  <Rect
    height="6"
    rx="3"
    width="410"
    x="0"
    y="56"
  />
  <Rect
    height="6"
    rx="3"
    width="380"
    x="0"
    y="72"
  />
  <Rect
    height="6"
    rx="3"
    width="178"
    x="0"
    y="88"
  />
  <Circle
    cx="20"
    cy="20"
    r="20"
  />
</NativeSvg>
`;
